File: frontend/src/lib/licitador.ts
export interface LegalDocumentItem {
  id: string;
  name: string;
  document_type: string;
  vigency_status: string;
  valid_until?: string | null;
  web_url?: string | null;
  source: string;
}

export interface CompanyLegalSummary {
  company_key: string;
  company_label: string;
  completeness_score: number;
  missing_profile_fields: string[];
  documents: LegalDocumentItem[];
  missing_documents: string[];
  expired_documents: string[];
  upcoming_documents: string[];
}

export interface LegalDocumentsDashboard {
  companies: CompanyLegalSummary[];
  total_documents: number;
  total_expired: number;
  total_missing: number;
}

export interface DgcpTemplateItem {
  id: string;
  name: string;
  source: string;
  document_category?: string | null;
  mime_type?: string | null;
  template_type?: string | null;
  web_url?: string | null;
  parent_path: string;
  fillable_fields_estimate: number;
}

export interface DgcpTemplatesDashboard {
  templates: DgcpTemplateItem[];
  total: number;
  by_category: Record<string, number>;
}

export interface PriceFileItem {
  id: string;
  name: string;
  supplier?: string | null;
  records: number;
  status: string;
  indexed_at?: string | null;
  parent_path: string;
}

export interface PriceIntelligenceDashboard {
  pending_files: PriceFileItem[];
  processed_files: PriceFileItem[];
  total_products: number;
  total_suppliers: number;
  lists_today: number;
  recent_errors: string[];
}

export interface TemplatePreviewField {
  key: string;
  label: string;
  value: string;
  source?: string;
}

export interface TemplatePreview {
  template_type: string;
  template_label: string;
  fields: TemplatePreviewField[];
  missing_fields: string[];
  generate_enabled: boolean;
}

export const COMPANY_OPTIONS = [
  { key: "justech", label: "JUSTECH" },
  { key: "just_office", label: "JUST OFFICE" },
  { key: "mf_plug_safe", label: "PLUG SAFE" },
  { key: "omni_solutions", label: "OMNI" },
] as const;

export const VIGENCY_STYLES: Record<string, string> = {
  vigente: "bg-emerald-500/15 text-emerald-700 border-emerald-500/30",
  proximo_a_vencer: "bg-amber-500/15 text-amber-800 border-amber-500/30",
  vencido: "bg-red-500/15 text-red-700 border-red-500/30",
  sin_fecha: "bg-muted text-muted-foreground",
};

export const TEMPLATE_TYPE_LABELS: Record<string, string> = {
  sncc: "Formulario SNCC",
  oferta_economica: "Oferta económica",
  declaracion_jurada: "Declaración jurada",
  autorizacion_fabricante: "Autorización fabricante",
  informacion_oferente: "Información oferente",
  carta: "Carta",
  dgcp: "Documento DGCP",
};

File: backend/app/schemas/licitador.py
"""Schemas — dashboards Licitador (legales, plantillas, precios)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class LegalDocumentItem(BaseModel):
    id: str
    name: str
    document_type: str
    vigency_status: str = "sin_fecha"
    valid_until: str | None = None
    web_url: str | None = None
    source: str = "knowledge"


class CompanyLegalSummary(BaseModel):
    company_key: str
    company_label: str
    completeness_score: int = 0
    missing_profile_fields: list[str] = Field(default_factory=list)
    documents: list[LegalDocumentItem] = Field(default_factory=list)
    missing_documents: list[str] = Field(default_factory=list)
    expired_documents: list[str] = Field(default_factory=list)
    upcoming_documents: list[str] = Field(default_factory=list)


class LegalDocumentsDashboardResponse(BaseModel):
    companies: list[CompanyLegalSummary] = Field(default_factory=list)
    total_documents: int = 0
    total_expired: int = 0
    total_missing: int = 0


class DgcpTemplateItem(BaseModel):
    id: str
    name: str
    source: str
    document_category: str | None = None
    mime_type: str | None = None
    template_type: str | None = None
    web_url: str | None = None
    parent_path: str = ""
    fillable_fields_estimate: int = 0


class DgcpTemplatesDashboardResponse(BaseModel):
    templates: list[DgcpTemplateItem] = Field(default_factory=list)
    total: int = 0
    by_category: dict[str, int] = Field(default_factory=dict)


class PriceFileItem(BaseModel):
    id: str
    name: str
    supplier: str | None = None
    records: int = 0
    status: str
    indexed_at: str | None = None
    parent_path: str = ""


class PriceIntelligenceDashboardResponse(BaseModel):
    pending_files: list[PriceFileItem] = Field(default_factory=list)
    processed_files: list[PriceFileItem] = Field(default_factory=list)
    total_products: int = 0
    total_suppliers: int = 0
    lists_today: int = 0
    recent_errors: list[str] = Field(default_factory=list)
